src/java.base/share/classes/sun/invoke/util/BytecodeDescriptor.java

```diff
@@ -148,6 +153,23 @@ private static Class<?> parseSig(String str, int[] i, int end, ClassLoader loade
         }
     }
 
+    private static final int CHECK_OFFSET = 32;
+    private static final long NON_IDENTIFIER_MASK = (1L << ('.' - CHECK_OFFSET))
+            | (1L << ('/' - CHECK_OFFSET))
+            | (1L << (';' - CHECK_OFFSET))
+            | (1L << ('[' - CHECK_OFFSET));
```

Comment:
Are we sure that these are the only 4 non-identifier chars we can see in the string?